Show selected program label in the filter on program report page load

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

Closes #651 

- Show selected program label in the filter on Programs Report page load

### Screenshots:

![Kapture 2021-05-28 at 12 58 32](https://user-images.githubusercontent.com/17420811/119931807-a9e04d80-bfb4-11eb-871f-c2c717e64f97.gif)

### Detailed test instructions:

1. Go to Programs Report page `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Freports%2Fprograms`
2. Select a paid program by the dropdown filter
3. Reload the page
4. The selected program label should show up in the filter on page load

### Changelog Note:

> Show selected program label in the filter on Programs Report page load

eason9487 committed:
Show selected program label in the filter on program report page load

tomalec reviewed (approved):
Tested locally, and reviewed the code. The original issue is solved, the program name appears once loaded. So I think we can merge it. 

However, we will not update the list of `autocompleter` options and program labels if they change in runtime, which I believe could be a problem in the future, see [the comment above](https://github.com/woocommerce/google-listings-and-ads/pull/683#issuecomment-853150869).

eason9487 committed:
Keep resolving changed campaign data in the program filter config

Address https://github.com/woocommerce/google-listings-and-ads/pull/683#issuecomment-853150869

eason9487 committed:
Rename waitNextData to waitForNextData

Address https://github.com/woocommerce/google-listings-and-ads/commit/92a68cb286ddd5399c3b49ca1f53b2dba5d4ed20#r51672057

tomalec committed:
Make sure every `createProgramsFilterConfig` test gets separate data instances.

To be safe agains some party violating immutability.

tomalec committed:
Use `async`/`await` in `createProgramsFilterConfig` tests.

To make the code more readable by shallower nesting.
Also, to wait for assertion in asynchronous flow, when we were asserting initial flow, but changing the config right after without waiting for it.
Addresses https://github.com/woocommerce/google-listings-and-ads/pull/683#commitcomment-51672039

Use `).toContain( * )` instead of `.includes( * ) ).toBe( true )` for more human friendly assertion errors.

tomalec committed:
Chaneg tested program ids to be `Number`s,

as that's what getLabels expect.
See `js/src/reports/programs/filter-config.js#L52:L52`

tomalec committed:
Add tests for `getLabels` from `createProgramsFilterConfig`

tomalec committed:
Merge pull request #727 from woocommerce/fix/651-filter-program-label-tomek

tomalec reviewed (approved):
FWIW, it still LGTM :)
